exareme2/algorithms/flower/logistic_regression/client.py

```diff
@@ -39,11 +40,23 @@ def evaluate(self, parameters, config):
 if __name__ == "__main__":
     model = LogisticRegression(penalty="l2", max_iter=1, warm_start=True)
     inputdata = get_input()
-    full_data = fetch_data(inputdata.data_model, inputdata.datasets, from_db=True)
+    full_data = fetch_client_data(inputdata)
     X_train, y_train = preprocess_data(inputdata, full_data)
     set_initial_params(model, X_train, full_data, inputdata)
 
     client = LogisticRegressionClient(model, X_train, y_train)
-    fl.client.start_client(
-        server_address=os.environ["SERVER_ADDRESS"], client=client.to_client()
-    )
+    max_retries = 6
+    for attempt in range(max_retries):
+        try:
+            fl.client.start_client(
+                server_address=os.environ["SERVER_ADDRESS"], client=client.to_client()
+            )
+            print("Connection successful on attempt", attempt + 1)
+            break
+        except Exception as e:
+            print(f"Connection attempt {attempt + 1} failed: {e}")
+            if attempt < max_retries - 1:
+                time.sleep(1)
+            else:
+                print("Max retries reached. Exiting.")
+                raise
```

Comment:
I think this should be a separate commit specifying the fix, the reason, etc.

Also `max_retries` could be an env variable, hard coded in the Dockerfile or somewhere else, whatever you prefer.

Finally this:
```
        except Exception as e:
            print(f"Connection attempt {attempt + 1} failed: {e}")
            if attempt < max_retries - 1:
                time.sleep(1)
            else:
                print("Max retries reached. Exiting.")
                raise
```
could be rewritten like:
```
        except Exception as e:
            if attempt >= max_retries - 1:
                print("Could not establish connection to the server.")
                raise ex
            print(f"Connection with the server failed. Attempt {attempt} failed: {e}")
            time.sleep(1) // USE GLOBAL VARIABLE
                
```

You can also start the `range()` from 1 instead of doing the +1 -1